### PR TITLE
Llorens day 09

### PR DESCRIPTION
**Pull request by JoseLlorensRipolles**
# [Your name]'s solution for day [day nº] challenge

## Description

> A simple description of your understanding of the problem may be useful.

## Solution

> Here you should paste the most important parts of your solution and explain them.
You can comment the rest of the code with revision comments.

## Tests battery

> This section can be the most interesting. Here, you can show the tests you have written for checking your solution.
You should explain why and which parts you are testing and how.

Happy Advent of Code!

### CODE REVIEW

File: llorens/day09/Day09Solver.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+def is_valid_char(j, sequence):
+    valid = True
+
+    for i in range(j-1, -1, -1):
```
Comment (Cotel) [inline llorens/day09/Day09Solver.py:4]: What? What kind of range is this generating?

File: llorens/day09/Day09Solver.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    sequence = sequence[:i] + sequence[j+1:]
+                    break
+                else:
+                    if sequence[j] != '!' and is_valid_char(j, sequence):
+                        num_valid_chars += 1
+                j += 1
+
+        i += 1
+
+    return sequence, num_valid_chars
```
Comment (Cotel) [inline llorens/day09/Day09Solver.py:33]: Is this returning a tuple?

File: llorens/day09/Day09Test.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        self.assertEqual(Solver.solve_part_01('{{<a!>},{<a!>},{<a!>},{<ab>}}'), 3, 'Part 1 Test 4')
+        self.assertEqual(Solver.solve_part_01(Reader.get_input("09")), 14204, 'Part 1 Input Test')
+
+    def test_part_02(self):
+
+        self.assertEqual(Solver.solve_part_02(Reader.get_input("09")), 6622, 'Part 2 Input Test')
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (Cotel) [inline llorens/day09/Day09Test.py:55]: Awesome tests 👌

**Review by Cotel — APPROVED** [review]
Many days haves passed since this so I couldn't review the code as good as I wanted to.

Anyway, I think this is a good solution and pretty readable 💯

File: llorens/day09/Day09Solver.py (hunk 1 above)
Comment (JoseLlorensRipolles) [reply llorens/day09/Day09Solver.py:4]: From j-1 to -1 and each iteration decreasing i by 1.

Range(From, To, Step)

**Pull request merged**